cmd/crud/main.go

```diff
@@ -58,21 +58,34 @@ func main() {
 		}
 		Trace struct {
 			Host         string        `default:"http://tracer:3002/v1/publish" envconfig:"HOST"`
-			BatchSize    int           `default:"1000" envconfig:"batch_size" envconfig:"BATCH_SIZE"`
-			SendInterval time.Duration `default:"15s" envconfig:"send_interval" envconfig:"SEND_INTERVAL"`
-			SendTimeout  time.Duration `default:"500ms" envconfig:"send_timeout" envconfig:"SEND_TIMEOUT"`
+			BatchSize    int           `default:"1000" envconfig:"BATCH_SIZE"`
+			SendInterval time.Duration `default:"15s" envconfig:"SEND_INTERVAL"`
+			SendTimeout  time.Duration `default:"500ms" envconfig:"SEND_TIMEOUT"`
 		}
 	}
 
 	if err := envconfig.Process("CRUD", &cfg); err != nil {
 		log.Fatalf("main : Parsing Config : %v", err)
 	}
 
+	if err := flag.Process(&cfg); err != nil {
+		if err != flag.ErrHelp {
+			log.Fatalf("main : Parsing Command Line : %v", err)
+		}
+		return // We displayed help.
+	}
+
+	// =========================================================================
+	// App Starting
+
+	log.Println("main : Started : Application Initializing")
+	defer log.Println("main : Completed")
+
 	cfgJSON, err := json.MarshalIndent(cfg, "", "    ")
 	if err != nil {
 		log.Fatalf("main : Marshalling Config to JSON : %v", err)
 	}
-	log.Printf("config : %v\n", string(cfgJSON))
+	log.Printf("main : Config : %v\n", string(cfgJSON))
```

Comment:
Is it ok to print the configuration to the logs? This essentially leaks credentials to a potentially unprotected log file.

Reply:
UGH!  We fixed this with our Config package by filtering certain keywords. Now this has moved back up to the app level. GREAT CATCH. All we can do it add a TODO, to check what config is being written?